### PR TITLE
Get contract address by version and customizable instantiate messages for tests

### PR DESCRIPTION
**Pull request by Buckram123**
Added some `croncat-integration-utils` features I was missing during implementing module for @AbstractSdk

### CODE REVIEW

File: integration-sdk/croncat-integration-utils/src/error.rs [hunk 1]
```diff
@@ -36,13 +36,17 @@ pub enum CronCatContractError {
     SerdeError { msg: String },
 
     #[error("Must attach funds for task creation")]
-    TaskCreationNoFunds,
+    TaskCreationNoFunds {},
 
-    #[error("No contract named {contract_name} on factory {factory_addr}")]
+    #[error("No contract named {contract_name} on factory {factory_addr} with {version} version")]
```
Comment (mikedotexe) [inline integration-sdk/croncat-integration-utils/src/error.rs:41]: very nice, thank you

File: integration-sdk/croncat-integration-utils/src/task_creation.rs [hunk 2]
```diff
@@ -23,16 +23,50 @@ pub fn get_latest_croncat_contract(
...
+                factory_addr: croncat_factory_address,
+                version: "latest".to_owned(),
+            })?;
+
+    Ok(contract_metadata.contract_addr)
+}
+
+/// Given the CronCat factory address and a version, returns the proper contract address if it exists.
+/// Use [get_latest_croncat_contract] to get latest version
+/// See [TASKS_NAME](crate::TASKS_NAME), [MANAGER_NAME](crate::MANAGER_NAME), and [AGENTS_NAME](crate::AGENTS_NAME)
```
Comment (mikedotexe) [inline integration-sdk/croncat-integration-utils/src/task_creation.rs:40]: the doc comments, they're so lovely. 🥺 

my MAN!

File: integration-sdk/croncat-integration-utils/src/task_creation.rs [hunk 3]
```diff
@@ -23,16 +23,50 @@ pub fn get_latest_croncat_contract(
...
+pub fn get_croncat_contract(
+    querier: &QuerierWrapper,
+    croncat_factory_address: Addr,
+    croncat_contract_name: String,
+    croncat_version: String,
+) -> Result<Addr, CronCatContractError> {
+    // Parse string to a version array, for example from "1.0" to [1,0]
+    let croncat_version_parsed = croncat_version
+        .split('.')
```
Comment (mikedotexe) [inline integration-sdk/croncat-integration-utils/src/task_creation.rs:49]: … so you *can* use single quotes in Rust

**Review by mikedotexe — APPROVED** [review]

**Comment by mikedotexe** [comment]
Sir, you may merge when ready! Thank you!

File: integration-sdk/croncat-integration-utils/src/task_creation.rs (hunk 3 above)
Comment (Buckram123) [reply integration-sdk/croncat-integration-utils/src/task_creation.rs:49]: Only for char, haha

**Comment by Buckram123** [comment]
> Sir, you may merge when ready! Thank you!

Did it, Sir. Thank you!

**Pull request merged**